compiler/src/dotty/tools/dotc/typer/VarianceChecker.scala

```diff
@@ -83,24 +83,21 @@ class VarianceChecker()(implicit ctx: Context) {
   private object Validator extends TypeAccumulator[Option[VarianceError]] {
     private var base: Symbol = _
 
-    /** Is no variance checking needed within definition of `base`? */
-    def ignoreVarianceIn(base: Symbol): Boolean = (
-         base.isTerm
-      || base.is(Package)
-      || base.isAllOf(PrivateLocal)
-    )
-
     /** The variance of a symbol occurrence of `tvar` seen at the level of the definition of `base`.
      *  The search proceeds from `base` to the owner of `tvar`.
      *  Initially the state is covariant, but it might change along the search.
      */
     def relativeVariance(tvar: Symbol, base: Symbol, v: Variance = Covariant): Variance = /*trace(i"relative variance of $tvar wrt $base, so far: $v")*/
-      if (base == tvar.owner) v
-      else if (base.is(Param) && base.owner.isTerm)
+      if base == tvar.owner then
+        v
+      else if base.is(Param) && base.owner.isTerm && !base.owner.isAllOf(PrivateLocal) then
         relativeVariance(tvar, paramOuter(base.owner), flip(v))
-      else if (ignoreVarianceIn(base.owner)) Bivariant
-      else if (base.isAliasType) relativeVariance(tvar, base.owner, Invariant)
-      else relativeVariance(tvar, base.owner, v)
+      else if base.owner.isTerm || base.owner.is(Package) || base.isAllOf(PrivateLocal) then
```

Comment:
by inlining `ignoreVarianceIn`, was it intentional to do `base.isAllOf(PrivateLocal)` and not `base.owner.isAllOf(PrivateLocal)`?

Reply:
Yes, the previous implementation was in fact wrong.